pkg/controllers/nodeclaim/disruption/drift.go

```diff
@@ -25,14 +25,17 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
+	corev1 "k8s.io/api/core/v1"
+
 	v1 "sigs.k8s.io/karpenter/pkg/apis/v1"
 	"sigs.k8s.io/karpenter/pkg/cloudprovider"
 	"sigs.k8s.io/karpenter/pkg/scheduling"
 )
 
 const (
-	NodePoolDrifted     cloudprovider.DriftReason = "NodePoolDrifted"
-	RequirementsDrifted cloudprovider.DriftReason = "RequirementsDrifted"
+	NodePoolDrifted           cloudprovider.DriftReason = "NodePoolDrifted"
+	RequirementsDrifted       cloudprovider.DriftReason = "RequirementsDrifted"
+	InstanceTypeNotFoundDrift cloudprovider.DriftReason = "InstanceTypeNotFound"
```

Comment:
```suggestion
	InstanceTypeNotFound cloudprovider.DriftReason = "InstanceTypeNotFound"
```

nit: The other types don't append drift at the end -- if you wanted to be more inline with K8s upstream, you could consider doing `NodePoolDriftedDriftReason`, `RequirementsDriftedDriftReason`, etc.